pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,311 @@
+"""
+pygmtlogo - Create and plot the PyGMT logo.
+The design of the logo is kindly provided by `@sfrooti <https://github.com/sfrooti>`_
+and consists of a visual and the wordmark "PyGMT".
+"""
+
+from pathlib import Path
+
+import numpy as np
+import pygmt
+
+
+def create_logo(color=True, theme="light", shape="circle", wordmark=True):  # noqa: PLR0915
+    """
+    Create the PyGMT logo using PyGMT.
+    The design of the logo is kindly provided by `@sfrooti <https://github.com/sfrooti>`_
+    and consists of a visual and the wordmark "PyGMT".
+
+    Parameters
+    ----------
+
+    color : bool
+        Set to ``True`` to use colors referring to Python (blue and yellow) and GMT
+        (red) [Default]. For ``False``, the logo is drawn in black and white.
+    theme : str
+        Use ``"light"`` for light mode (i.e., a white background) [Default] and
+        ``"dark"`` for dark mode (i.e., a darkgray [gray20] background).
+    shape : str
+        Shape of the visual. Use ``"circle"`` for a circle shape [Default] or
+        ``"hexagon"`` for a hexagon shape.
+    wordmark : bool, str
+        Add the wordmark "PyGMT" and adjust its orientation relative to the visual. Set
+        to ``True`` or ``"horizontal"``, to add the wordmark at the right side of the
+        visual [Default]. Use ``"vertical"`` to place the wordmark below the visual
+        and ``False`` to add no wordmark.
+    """
+
+    # Helpful definitions
+    size = 4
+    region = [-size, size] * 2
+    projection = "x1c"
+
+    # Radii
+    r0, r1, r2, r3, r4, r5 = size * np.array(
+        [1.0, 0.875, 0.58125, 0.4625, 0.4125, 0.29375]
+    )
+    thick = r0 - r1  # thick pen in centimeters
+    thin = thick / 3.0  # thin pen in centimeters
+
+    # Rotation around z (vertical) axis placed in the center
+    # Has to be applied to each plotting command, up on second call set to True
+    perspective = "30+w0/0"  # by 30 degrees ccw
+
+    # Define colors
+    color_light = "white"
+    color_dark = "gray20"
+
+    blue = "48/105/152"  # Python blue
+    yellow = "255/212/59"  # Python yellow
+    red = "238/86/52"  # GMT red
+    if not color:
+        blue = yellow = red = color_dark
+        if theme == "dark":
+            blue = yellow = red = color_light
+
+    # Background and wordmark
+    match theme:
+        case "light":
+            color_bg = color_light
+            color_py = blue
+            color_gmt = color_dark
+        case "dark":
+            color_bg = color_dark
+            color_py = yellow
+            color_gmt = color_light
+
+    # Define shape
+    match shape:
+        case "circle":
+            symbol = "c"  # circle
+            size_shape = r0 + r1  # radius
+            size_shape_add = r0 - r1
+            hex_factor = 1
```

Comment:
Instead of defining variables like `hex_factor`/`y_vertline`/`y_arrow`. Can we just define different radii for circle and hexagon shapes?